src/player.cpp

```diff
@@ -856,8 +873,10 @@ static void playerUpdateHandler(TMario *player, JDrama::TGraphics *graphics) {
 
     auto* params = Player::getData(player);
 
-    if(params->mMActor) {
-        params->mMActor->frameUpdate();
+    if(params->mMActor[0]) {
```

Comment:
Change this to check for each MActor individually within the loop because that's safer and also more flexible.